org.lflang/src/org/lflang/cli/Lfc.java

```diff
@@ -52,83 +56,98 @@ public Lfc() {
-        SCHEDULER(BuildParm.SCHEDULER, "s", true, false, true),
-        THREADING(BuildParm.THREADING, "t", true, false, true),
-        WORKERS(BuildParm.WORKERS, "w", true, false, true);
-
-        /**
-         * The corresponding Apache CLI Option object.
-         */
-        public final Option option;
-
-        /**
-         * Whether to pass this option to the code generator.
-         */
-        public final boolean passOn;
-
-        /**
-         * Construct a new CLIOption.
-         *
-         * @param parameter   The build parameter that this CLI parameter corresponds to.
-         * @param shorthand   The single-character switch to use for this option. E.g.:
-         *                    "-c" for "--clean".
-         * @param hasArg      Whether this option has an argument. E.g.:
-         *                    "--foo bar" where "bar" is the argument value.
-         * @param isReq       Whether this option is required. If it is
-         *                    required but not specified a menu is shown.
-         * @param passOn      Whether to pass this option as a property
-         *                    to the code generator.
-         */
-        CLIOption(BuildParm parameter, String shorthand, boolean hasArg, boolean isReq, boolean passOn) {
-            this.option = new Option(shorthand, parameter.getKey(), hasArg, parameter.description);
-            option.setRequired(isReq);
-            this.passOn = passOn;
-        }
 
-        /**
-         * Create an Apache Commons CLI Options object and add all the options.
-         *
-         * @return Options object that includes all the options in this enum.
-         */
-        public static Options getOptions() {
-            Options opts = new Options();
-            Arrays.asList(CLIOption.values()).forEach(o -> opts.addOption(o.option));
-            return opts;
-        }
+    @Option(
+        names = "--build-type",
+        description = "The build type to use.")
+    private String buildType;
 
-        /**
-         * Return a list of options that are to be passed on to the code
-         * generator.
-         *
-         * @return List of options that must be passed on to the code gen stage.
-         */
-        public static List<Option> getPassedOptions() {
-            return Arrays.stream(CLIOption.values())
-                .filter(opt -> opt.passOn).map(opt -> opt.option)
-                .collect(Collectors.toList());
-        }
-    }
+    @Option(
+        names = {"-c", "--clean"},
+        arity = "0",
+        description = "Clean before building.")
+    private boolean clean;
+
+    @Option(
+        names = "--target-compiler",
+        description = "Target compiler to invoke.")
+    private String targetCompiler;
+
+    @Option(
+        names = "--external-runtime-path",
+        description = "Specify an external runtime library to be used by the"
+                    + " compiled binary.")
+    private String externalRuntimePath;
```

Comment:
Thanks for the suggestion! Picocli does support `java.nio.file.Path` directly - I just changed the input files parameter and the `outputPath` option to be interpreted as `Path`s instead of strings in Lfc/Lff. I've left all other options as Strings (or booleans to distinguish options that take no args) though, because later on they're passed into `LFGeneratorContext` through a `java.utils.Properties` object, which expects all option values to be strings.

Reply:
Is there a fundamental reason for using `java.utils.Properties`? Maybe we should just create our own record class that passes the options in the correct types. Seems like converting to String and then back to the data type is redundant. This doesn't need to happen in this PR though, we can also leave for future work.

Reply:
Tbh even if the options need to be converted to a string later, it is cleaner to use the correct data types as early as possible in the data flow. It allows us to take advantage of built-in validation and improves the CLI help. Converting to a string is also very easy.

Reply:
That's true, I'll use the correct data types (with custom type convertors wherever needed) to make sure all option values are validated at the entry point of the CLI.

Reply:
Looks like you missed the option here, it is still a String and not a Path.

Reply:
Just fixed it! Also, I wanted to write custom type converters/validators (e.g. checking if a path actually exists), but that would require me to write a separate class implementing the `picocli.CommandLine.ITypeConverter` interface. However, our current validation logic uses protected CliBase members, like `io` and `reporter`, which I don't think I'll be able to access from the separate type converter class. Do you have any ideas for workarounds?